modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/VolatilitySwaptionProductPricer.java

```diff
@@ -126,6 +132,19 @@ public double impliedVolatility(
     }
   }
 
+  //-------------------------------------------------------------------------
+  /**
+   * Provides the forward rate.
+   * This is the par rate for the forward starting swap that is the underlying of the swaption.
+   *
+   * @param swaption the swaption
+   * @param ratesProvider the rates provider
+   * @return the forward rate
+   */
+  public static double forwardRate(ResolvedSwaption swaption, RatesProvider ratesProvider) {
+    return SWAP_PRICER.parRate(swaption.getUnderlying(), ratesProvider);
```

Comment:
@jodastephen Is it not preferable to follow the pattern of the other methods in the product by delegating to the correct pricer (`cashParYieldPricer` or `physicalPricer`) and then calling the swap pricer there? Just seems a bit inconsistent with all the other methods and pricers and is easier to follow.

Reply:
Yes, that makes sense